include/boost/geometry/formulas/karney_direct.hpp

```diff
@@ -0,0 +1,290 @@
+                                (sin_tau1 * cos_tau12 + cos_tau1 * sin_tau12,
+                                 cos_tau1 * cos_tau12 - sin_tau1 * sin_tau12,
+                                 coeffs_C1p);
+
+        CT const sigma12 = tau12 - (B12 - B11);
+        CT const sin_sigma12 = sin(sigma12);
+        CT const cos_sigma12 = cos(sigma12);
+
+        CT const sin_sigma2
+            = sin_sigma1 * cos_sigma12 + cos_sigma1 * sin_sigma12;
+        CT const cos_sigma2
+            = cos_sigma1 * cos_sigma12 - sin_sigma1 * sin_sigma12;
+
+        if (BOOST_GEOMETRY_CONDITION(CalcRevAzimuth))
+        {
+            CT const sin_alpha2 = sin_alpha0;
+            CT const cos_alpha2 = cos_alpha0 * cos_sigma2;
+
+            result.reverse_azimuth = atan2(sin_alpha2, cos_alpha2);
+
+            // Convert the angle to radians.
+            result.reverse_azimuth /= math::d2r<T>();
+        }
+
+        if (BOOST_GEOMETRY_CONDITION(CalcCoordinates))
+        {
+            // Find the latitude at the second point.
+            CT const sin_beta2
+                = cos_alpha0 * sin_sigma2;
+            CT const cos_beta2
+                = boost::math::hypot(sin_alpha0, cos_alpha0 * cos_sigma2);
+
+            result.lat2 = atan2(sin_beta2, one_minus_f * cos_beta2);
+
+            // Convert the coordinate to radians.
+            result.lat2 /= math::d2r<T>();
+
+            // Find the longitude at the second point.
+            CT const sin_omega2 = sin_alpha0 * sin_sigma2;
+            CT const cos_omega2 = cos_sigma2;
+
+            CT const omega12 = atan2(sin_omega2 * cos_omega1 - cos_omega2 * sin_omega1,
+                                     cos_omega2 * cos_omega1 + sin_omega2 * sin_omega1);
+
+            CT coeffs_A3[SeriesOrder];
+            series_expansion::evaluate_coeffs_A3<CT, SeriesOrder>(n, coeffs_A3);
+
+            CT const A3 = math::horner_evaluate(epsilon, coeffs_A3, coeffs_A3 + SeriesOrder);
+            CT const A3c = -f * sin_alpha0 * A3;
+
+            // Compute the size of coefficient array.
+            size_t const coeffs_C3_size = (SeriesOrder * (SeriesOrder - 1)) / 2;
+            CT coeffs_C3x[coeffs_C3_size];
+            series_expansion::evaluate_coeffs_C3x<CT, SeriesOrder>(n, coeffs_C3x);
+
+            // Evaluate C3 coefficients.
+            CT coeffs_C3[SeriesOrder];
+            series_expansion::evaluate_coeffs_C3<CT, SeriesOrder>(epsilon, coeffs_C3, coeffs_C3x);
+
+            CT const B31 =
+                series_expansion::sin_cos_series<CT, SeriesOrder>(sin_sigma1, cos_sigma1, coeffs_C3);
+
+            CT const lam12 = omega12 + A3c *
+                             (sigma12 + (series_expansion::sin_cos_series<CT, SeriesOrder>
+                                                           (sin_sigma2,
+                                                            cos_sigma2,
+                                                            coeffs_C3) - B31));
```

Comment:
I think the `sin_cos_series` function is fixed now. Previously,
I was passing `SeriesOrder` for all coefficients, but now the
size is queried from `boost::array`'s `static_size` member.

So, if the container size is `9`, `n` is set to `(9 - 1)`. This line:
```
CT k0 = n & 1 ? coeffs[--index] : 0;
```
checks if `n` is odd (not in this case), so `k0` is set to `0`.

The loop runs 4 times because of `n /= 2`.
So, in total 8 elements are accessed (excluding the first one).

If the container size is `8`, then all the container elements are
used for computing the result.